go-controller/pkg/ovn/ovn.go

```diff
@@ -406,35 +406,46 @@ func (oc *DefaultNetworkController) syncNodeGateway(node *kapi.Node, hostSubnets
 
 // gatewayChanged() compares old annotations to new and returns true if something has changed.
 func gatewayChanged(oldNode, newNode *kapi.Node) bool {
-	oldL3GatewayConfig, _ := util.ParseNodeL3GatewayAnnotation(oldNode)
-	l3GatewayConfig, _ := util.ParseNodeL3GatewayAnnotation(newNode)
-	return !reflect.DeepEqual(oldL3GatewayConfig, l3GatewayConfig)
+	return oldNode.Annotations[util.OvnNodeL3GatewayConfig] != newNode.Annotations[util.OvnNodeL3GatewayConfig] ||
+		oldNode.Annotations[util.OvnNodeChassisID] != newNode.Annotations[util.OvnNodeChassisID]
 }
 
 // hostCIDRsChanged compares old annotations to new and returns true if the something has changed.
 func hostCIDRsChanged(oldNode, newNode *kapi.Node) bool {
-	oldAddrs, _ := util.ParseNodeHostCIDRs(oldNode)
-	Addrs, _ := util.ParseNodeHostCIDRs(newNode)
-	return !oldAddrs.Equal(Addrs)
+	return util.NodeHostCIDRsAnnotationChanged(oldNode, newNode)
 }
 
 // macAddressChanged() compares old annotations to new and returns true if something has changed.
 func macAddressChanged(oldNode, node *kapi.Node, netName string) bool {
-	oldMacAddress, _ := util.ParseNodeManagementPortMACAddresses(oldNode, netName)
-	macAddress, _ := util.ParseNodeManagementPortMACAddresses(node, netName)
-	return !bytes.Equal(oldMacAddress, macAddress)
+	var oldMacAddress, newMacAddress map[string]json.RawMessage
+
+	if err := json.Unmarshal([]byte(oldNode.Annotations[util.OvnNodeManagementPortMacAddresses]), &oldMacAddress); err != nil {
+		klog.Errorf("Failed to unmarshal old node %s annotation: %v", oldNode.Name, err)
+		return true
+	}
+	if err := json.Unmarshal([]byte(node.Annotations[util.OvnNodeManagementPortMacAddresses]), &newMacAddress); err != nil {
+		klog.Errorf("Failed to unmarshal new node %s annotation: %v", node.Name, err)
```

Comment:
Not sure how useful it is going to be to return true when the new value fails to parse. Assuming that processing the corresponding event will also fail and we get a barrage of useless retries. Same comment for other instances. For your consideration, not asking fro change.

Reply:
We were ignoring errors previously but I agree that it doesn't make sense to return true if we failed to parse.